add GC b&r job options docs (#4771)
treeverse/lakeFS: docs/howto/garbage-collection.md

* Backup: copy expired objects from your repository's storage namespace to an external location before running GC in [sweep only mode](#sweep-only-mode).  
* Restore: copy objects that were hard-deleted by GC from an external location you used for saving your backup into your repository's storage namespace.

**Notes**
* GC backup and restore is available from version 0.5.2 of lakeFS Spark client.

> * This will only work with Hadoop versions > 3.1.3, and so require a Hadoop 3 client.

Reviewer: ```suggestion
* It is compatible with Hadoop API versions 3.1.3 and higher.
```
?

Author: done

> * Note that the utility is not fast due to distcp performance limitations. You may prefer backup your whole storage namespace with [AzCopy](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10) / [aws cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) / [rclone](https://rclone.org/).

Reviewer: ```suggestion
* Note that the utility is not fast due to distcp performance limitations. You may prefer to backup your whole storage namespace with [AzCopy](https://learn.microsoft.com/en-us/azure/storage/common/storage-use-azcopy-v10) / [aws cp](https://docs.aws.amazon.com/cli/latest/reference/s3/cp.html) / [rclone](https://rclone.org/).
```

Author: done
